cmd/rpcdaemon/commands/tracing.go

```diff
@@ -200,6 +201,21 @@ func (api *PrivateDebugAPIImpl) TraceTransaction(ctx context.Context, hash commo
 		return nil
 	}
 
+	if chainConfig.IsOptimismPreBedrock(blockNum) {
+		if api.historicalRPCService != nil {
+			treeResult := &GethTrace{}
+			if err := api.historicalRPCService.CallContext(ctx, treeResult, "debug_traceTransaction", hash, config); err != nil {
+				return fmt.Errorf("historical backend failed: %w", err)
+			}
+			result, err := json.Marshal(treeResult)
+			if err != nil {
+				return err
+			}
+			stream.WriteRaw(string(result))
```

Comment:
I'd assume we want to return `nil` or capture an error here? (As written, I think we always fall through to return `rpc.ErrNoHistoricalFallback`)

Reply:
Good catch! Fixed